packages/core/src/vuePlugins/actions/useExtendedMasterportalapiMarkers/index.ts

```diff
@@ -7,28 +7,47 @@ import {
 } from '@polar/lib-custom-types'
 import RenderFeature from 'ol/render/Feature'
 import { isVisible } from '@polar/lib-invisible-style'
-import { getHoveredStyle, getSelectedStyle } from '../../utils/markers'
-import { resolveClusterClick } from '../../utils/resolveClusterClick'
+import { Store } from 'vuex'
+import { Point } from 'ol/geom'
+import { easeOut } from 'ol/easing'
+import { getHoveredStyle, getSelectedStyle } from '../../../utils/markers'
+import { resolveClusterClick } from '../../../utils/resolveClusterClick'
+import { setLayerId } from './setLayerId'
+
+// TODO pull file apart (after changes in other PR are through)
 
 let lastClickEvent: MapBrowserEvent<MouseEvent> | null = null
 
 // local copies
 let hovered: Feature | null = null
 let selected: Feature | null = null
 
+const center = (map: Map, selected: Feature | null) => {
+  if (selected !== null) {
+    map.getView().animate({
+      center: (selected.getGeometry() as Point).getCoordinates(),
+      duration: 400,
+      easing: easeOut,
+    })
+  }
+}
+
 // eslint-disable-next-line max-lines-per-function
 export function useExtendedMasterportalapiMarkers(
-  { getters, commit }: PolarActionContext<CoreState, CoreGetters>,
+  this: Store<CoreState>,
```

Comment:
Generally, `PolarActionHandler` would be the type to use for the whole function but as this seems a bit weird to type, adding `PolarStore` would be better

Reply:
https://github.com/Dataport/polar/pull/43/commits/d0a21113aef5e373990d9fc9efc3be512bea4a08 Agreed.